feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.ts

```diff
@@ -100,17 +101,23 @@ export class CheckoutDeliveryModeComponent {
     protected activeCartFacade: ActiveCartFacade
   ) {}
 
-  changeMode(code: string | undefined): void {
+  changeMode(code: string | undefined, event?: Event): void {
     if (!code) {
       return;
     }
-
+    const lastFocusedId = (<HTMLElement>event?.target)?.id;
     this.busy$.next(true);
-
     this.checkoutDeliveryModesFacade.setDeliveryMode(code).subscribe({
       complete: () => this.onSuccess(),
       error: () => this.onError(),
     });
+
+    const isTriggeredByKeyboard = (<MouseEvent>event)?.screenX === 0;
+    if (isTriggeredByKeyboard) {
+      this.restoreFocus(lastFocusedId, code);
+      return;
+    }
+    this.mode.setValue({ deliveryModeId: code });
```

Comment:
Should we hide this section behind a feature flag?

Reply:
Feature flag added ;)